### PR TITLE
Docs (GraphQL): 21.03 feature documentation for lambda webhooks

### PR DESCRIPTION
**Pull request by vardhanapoorv**
Docs regarding lambda webhooks.
Fixes DOC-199

### CODE REVIEW

File: content/graphql/lambda/webhook.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+title = "Lambda Webhooks"
+description = "Ready to use lambdas for webhooks? This documentation takes you through the schemas, resolvers, and examples."
+weight = 5
+[menu.main]
+    parent = "lambda"
++++
+
+### Schema
+
+To set up a lambda webhook, first you need to define it in your GraphQL schema by using the `@lambdaOnMutate` directive along with the mutation events (add/update/delete) you want to listen on.
```
Comment (bucanero) [inline content/graphql/lambda/webhook.md:11]:
```suggestion
To set up a lambda webhook, you need to define it in your GraphQL schema by using the `@lambdaOnMutate` directive along with the mutation events (`add`/`update`/`delete`) you want to listen on.
```

File: content/graphql/lambda/webhook.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+For example, to define a lambda webhook for all mutation events (add/update/delete) on any `Author` object:
```
Comment (bucanero) [inline content/graphql/lambda/webhook.md:13]:
```suggestion
For example, to define a lambda webhook for all mutation events (`add`/`update`/`delete`) on any `Author` object:
```

File: content/graphql/lambda/webhook.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+self.addWebHookResolvers({
+    "Author.add": addAuthorWebhook,
+    "Author.update": updateAuthorWebhook,
+    "Author.delete": deleteAuthorWebhook,
+})
+```
+
+### Example
+
+Finally, if you execute a `addAuthor` mutation the resolver `addAuthorWebhook` mapped to `add` operation will get triggered :
```
Comment (bucanero) [inline content/graphql/lambda/webhook.md:84]:
```suggestion
Finally, if you execute an `addAuthor` mutation the resolver `addAuthorWebhook` mapped to the `add` operation will get triggered:
```

File: content/graphql/lambda/webhook.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+### Resolver
+
+Once the schema is ready, you can define your JavaScript functions and add those as resolvers in your JS source code. 
+To add the resolvers you should use the `addWebHookResolvers`method.
+
+{{% notice "note" %}}
+A Lambda Webhook resolver can use a combination of `event`, `dql`, `graphql` or `authHeader` inside the function.
+{{% /notice %}}
+
+You have access to the `event` object within the resolver. The definition of `event` is something as follows:
```
Comment (bucanero) [inline content/graphql/lambda/webhook.md:33]:
```suggestion
#### Event object

You also have access to the `event` object within the resolver. The definition of `event` is as follows:
```

File: content/graphql/lambda/webhook.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      "rootUIDs": [<list-of-UIDs-of-root-nodes-for-which-somethig-was-set/removed-in-this-mutation>],
+      "setPatch": <TypePatch: the object that was received as the patch for set>,
+      "removePatch": <TypePatch: the object that was received as the patch for remove>
+    },
+    "delete": {
+      "rootUIDs": [<list-of-UIDs-of-root-nodes-which-were-deleted-in-this-mutation>]
+    }
+```
+
+For example, to define the JavaScript lambda functions for each mutation event for which `@lambdaOnMutate` is enabled and add those as resolvers:
```
Comment (bucanero) [inline content/graphql/lambda/webhook.md:54]:
```suggestion
#### Resolver examples

For example, to define JavaScript lambda functions for each mutation event for which `@lambdaOnMutate` is enabled and add those as resolvers:
```

**Review by bucanero — COMMENTED** [review]
I added some suggestions 👍

**Review by bucanero — APPROVED** [review]

**Comment by bucanero** [comment]
@abhimanyusinghgaur could you review this PR? thanks!

File: content/graphql/lambda/webhook.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+To add the resolvers you should use the `addWebHookResolvers`method.
+
+{{% notice "note" %}}
+A Lambda Webhook resolver can use a combination of `event`, `dql`, `graphql` or `authHeader` inside the function.
+{{% /notice %}}
+
+#### Event object
+
+You also have access to the `event` object within the resolver. The definition of `event` is as follows:
+
```
Comment (abhimanyusinghgaur) [inline content/graphql/lambda/webhook.md:36]: we can mention that based on the value of `operation`, only one of add, update or delete keys will be present inside the `event` object. i.e., there will always be `event[event["operation"]]` and nothing else.

**Review by abhimanyusinghgaur — APPROVED** [review]
Looks perfect!
Should we also add the new directive `@lambdaOnMutate` on the directives index here: https://dgraph.io/docs/graphql/directives/

File: content/graphql/lambda/webhook.md (hunk 6 above)
Comment (vardhanapoorv) [reply content/graphql/lambda/webhook.md:36]: Done

File: content/graphql/directives/index.md [hunk 7]
```diff
@@ -102,3 +102,9 @@ Reference: [Skip directive](/graphql/queries/skip-include)
 `@withSubscription` directive when applied on a type, generates subsciption queries for it.
 
 Reference: [Subscriptions](/graphql/subscriptions)
+
+### @lambdaOnMutate
+
+The `@lambdaOnMutate` directive allows you to listen on mutation events(`add`/`update`/`delete`). The `@lambdaOnMutate` depending on the mutation events defined and on occurence of a mutation event, triggers the appropriate lambda function implemented on a given lambda server.
```
Comment (bucanero) [inline content/graphql/directives/index.md:108]:
```suggestion
The `@lambdaOnMutate` directive allows you to listen to mutation events(`add`/`update`/`delete`). Depending on the defined events and the occurrence of a mutation event, `@lambdaOnMutate` triggers the appropriate lambda function implemented on a given lambda server.
```

**Pull request merged**